applications/TrilinosApplication/trilinos_space.h

```diff
@@ -656,100 +658,6 @@ class TrilinosSpace
         rX.PutScalar(0.0);
     }
 
-    /// TODO: creating the the calculating reaction version
-    // 	template<class TOtherMatrixType, class TEquationIdVectorType>
-
-    /**
-     * @brief Assembles the LHS of the system
-     * @param rA The LHS matrix
-     * @param rLHSContribution The contribution to the LHS
-     * @param rEquationId The equation ids
-     */
-    inline static void AssembleLHS(
```

Comment:
The Assemble{LHS,RHS} functions are part of the space functionalities (the Ublas one has them too), I would leave them here

Reply:
To make myself clear, since we are having this discussion over several PRs: In my opinion, functions already existing on both spaces are ok where they are. New functions, or added dependencies to distributed-memory specifics (such as references to global/local indexing, fill_complete and other trilinos specifics) I would move to the utility class.

I realize that this is hard to do here (the Mult functions are a bit of a problem in this regard), so take my comments with a grain of salt.

Reply:
Should be Okay now